mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -177,35 +182,36 @@ def dialog_renderer(self):
         # Fall back to main language
         return self.dialog_renderers.get(self._core_lang)
 
+    @property
+    def settings_path(self):
+        return join(xdg.BaseDirectory.save_config_path(BASE_FOLDER, 'skills', self.skill_id), 'settings.json')
+
+    @property
+    def settings_write_path(self):
+        # hopefully nothing in the outside world ever noticed this was not underscore prefixed
+        LOG.warning("self.settings_write_path has been deprecated! do not use it, it is an internal variable")
+        return xdg.BaseDirectory.save_config_path(BASE_FOLDER, 'skills', self.skill_id)
```

Comment:
comments from #9 by @NeonDaniel 
_________________
Consider consolidating skill FS paths as discussed https://github.com/MycroftAI/mycroft-core/issues/2912

Also, this is a breaking change in NeonCore
https://github.com/NeonGeckoCom/neon-skill-utils/blob/dc93b9be163af5339361e03d85bd79bf67857e45/neon_utils/skills/mycroft_skill.py#L75

Reply:
that would go against the XDG standard we have been pushing towards.... imho skill settings should only live in the XDG path, even in neon-core

still can't make setting_write_path a property, will keep it a variable and add deprecation warnings at least

Reply:
it should no longer be a breaking change, but i still marked it for deprecation

Reply:
Should skill FS be the same XDG path as skill settings? Per XDG, it's probably more accurately `data` and not `config`, but from a user/dev perspective I think it makes sense for all the skill support files to be stored in a single directory

Reply:
I'm not convinced we must force it into a single directory, i think following the specs the whole unix world follows end up being less confusing...

I think it makes sense to separate data from config, the same reason we dont have skills and mycroft.conf in the same location

In this instance i think its less confusing if all configs go into same place all other app/core configs go and the same for all application data

I see self.file_system as skill internal data, managed by the skill, unrelated to skill settings which i view as managed/shared by core, users and external applications

One other point is that we lose compatibility with mycroft-core if they don't adopt the same change, causing even further confusion among users which kinda defeats the point.

I can be convinced otherwise, interested to hear the broader community's opinion

Reply:
this PR remains compatible (i hope! please test and review under neon) with current mycroft-core behavior, so the path change can be revisited later

I can change the deprecation message until a decision is made, if this remains fully backwards compatible we can merge in the meantime for the init fixes because #9 depends on it

Reply:
I agree that the end result of where things go should be determined upstream in mycroft-core. Since this maintains compatibility there, I suppose this makes sense (for now). Any breakage in Neon Core should probably be fixed there as these changes look good and appear to be compatible